eurocropsml/acquisition/region.py

```diff
@@ -36,142 +36,167 @@ def add_nuts_regions(
             This depends on the satellite.
 
     Raises:
-        ValueError: If NUTS-files are not available.
+        FileNotFoundError: If NUTS-files are not available.
+        FileNotFoundError: If the merged clipped file from clipping module does not exist.
 
     """
     url: str = (
         "https://ec.europa.eu/eurostat/de/web/gisco/geodata/"
         "statistical-units/territorial-units-statistics"
     )
 
-    label_dir = raw_data_dir.joinpath("labels")
-    geom_dir = raw_data_dir.joinpath("geometries")
+    shapefile: gpd.GeoDataFrame = pyogrio.read_dataframe(shape_dir)
 
-    if not (
-        label_dir.joinpath(f"{config.country}_labels.parquet").exists()
-        and final_output_dir.joinpath(f"{config.country}.parquet").exists()
-        and geom_dir.joinpath(f"{config.country}.geojson").exists()
-    ):
-        shapefile: gpd.GeoDataFrame = pyogrio.read_dataframe(shape_dir)
+    # fix invalid geometries
+    shapefile["geometry"] = shapefile["geometry"].buffer(0)
+
+    crs: str = str(shapefile.crs).split(":")[-1]
 
-        # fix invalid geometries
-        shapefile["geometry"] = shapefile["geometry"].buffer(0)
+    # get nuts regions
+    nuts_region_filename = f"NUTS_RG_01M_{config.year}_{crs}.geojson"
+    nuts_regions_file = nuts_dir.joinpath(nuts_region_filename)
 
-        crs: str = str(shapefile.crs).split(":")[-1]
+    if not nuts_dir.exists() or len(list((nuts_dir.iterdir()))) < 3:
+        nuts_dir.mkdir(exist_ok=True, parents=True)
+        _nuts_region_downloader(
+            url, nuts_dir, crs, config.year, [file.name for file in nuts_dir.iterdir()]
+        )
 
-        # get nuts regions
+    if not any(nuts_dir.iterdir()):
+        raise FileNotFoundError(
+            "There are no NUTS region files available."
+            f" Please download them manually from {url}."
+        )
+    try:
+        nuts: gpd.GeoDataFrame = pyogrio.read_dataframe(nuts_regions_file)
+
+    except Exception:
```

Comment:
Can we be more explicit about what type of Exception are we expecting to catch here?

Reply:
Replace by FileNotFoundError.